src/libcore/ops.rs

```diff
@@ -102,6 +102,9 @@ pub trait Drop {
     ///
     /// After this function is over, the memory of `self` will be deallocated.
     ///
+    /// This function cannot be called explicitly. This is compiler error
+    /// [0040](https://doc.rust-lang.org/error-index.html#E0040).
```

Comment:
Added a mention. Out of curiosity, do you know why calling the destructor is discouraged when it's easy enough to call  `std::mem::drop` (which would be easy enough to write even if it didn't exist)?